misc/AL2023-Dockerfile

```diff
@@ -19,6 +19,12 @@ RUN dnf install -y \
     automake \
     autoconf
 
+# Some basic tools that are typical during builds
+RUN dnf install -y --allowerasing \
+    gnupg2 \
+    gcc \
+    make
+
```

Comment:
Can we add the same to AL2 and Fedora Dockerfiles? Otherwise this commit is more of a quick hack to make stuff work only on AL2023.